File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.cpp
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {
        int n = nums.size();
        vector<int> ans, left, right;
        int i = 0, j = n - 1, cnt = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] < pivot) left.push_back(nums[i]);
            else if (nums[i] > pivot) right.push_back(nums[i]);
            else cnt++;
        }
        for (auto v : left) {
            ans.push_back(v);
        }
        for (int i = 0; i < cnt; i++) {
            ans.push_back(pivot);
        }
        for (auto v : right) {
            ans.push_back(v);
        }
        return ans;
    }
};

File: 2161-partition-array-according-to-given-pivot/2161-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int n = nums.length;
        ArrayList<Integer> right = new ArrayList<>();
        int[] ans = new int[n];
        int ind = 0;
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            if (nums[i] < pivot) {
                ans[ind++] = nums[i];
            } else if (nums[i] > pivot) {
                right.add(nums[i]);
            } else {
                cnt++;
            }
        }
        while (cnt > 0) {
            ans[ind++] = pivot;
            cnt--;
        }
        for (int num : right) {
            ans[ind++] = num;
        }
        return ans;
    }
}